chia/rpc/wallet_rpc_api.py

```diff
@@ -1111,10 +1138,20 @@ async def did_get_recovery_list(self, request):
         return {
             "success": True,
             "wallet_id": wallet_id,
-            "recover_list": recover_hex_list,
+            "recovery_list": recover_hex_list,
             "num_required": wallet.did_info.num_of_backup_ids_needed,
         }
 
+    async def did_get_metadata(self, request):
+        wallet_id = int(request["wallet_id"])
+        wallet: DIDWallet = self.service.wallet_state_manager.wallets[wallet_id]
+        metadata = json.loads(wallet.did_info.metadata)
```

Comment:
Is there a basic schema/shape that we're expecting the metadata to conform to? Array of simple string --> string key value pairs, for instance? I'd be careful with accepting whatever JSON data the user provides.

Reply:
I would also recommend wrapping the JSON parsing in a try:except: so that we can return an error stating that the JSON was invalid. Otherwise, the user is going to see whatever exception was thrown by the JSON parser.